daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -718,3 +741,196 @@ class UnparseResult(dp: DataProcessor, ustate: UState)
     encodingInfo.knownEncodingName
   }
 }
+
+class DaffodilXMLReader(dp: DataProcessor) extends DFDL.DaffodilXMLReader {
+  private var contentHandler: ContentHandler = _
+  private var errorHandler: ErrorHandler = _
+  private var dtdHandler: DTDHandler = _
+  private var entityResolver: EntityResolver = _
+  private val saxNamespaceFeature = "http://xml.org/sax/features/namespaces"
+  private val saxNamespacePrefixFeature = "http://xml.org/sax/features/namespace-prefixes"
+  private val saxSupportedProperties = Seq(s"(${XMLUtils.DAFFODIL_SAX_URN_ROOT}:ParseResult",
+    s"${XMLUtils.DAFFODIL_SAX_URN_ROOT}:BlobDirectory",
+    s"${XMLUtils.DAFFODIL_SAX_URN_ROOT}:BlobPrefix",
+    s"${XMLUtils.DAFFODIL_SAX_URN_ROOT}:BlobSuffix")
+  var saxParseResultProperty: ParseResult = _
+  var saxBlobDirectoryProperty: Path = Paths.get(System.getProperty("java.io.tmpdir"))
+  var saxBlobPrefixProperty: String = "daffodil-sax-"
+  var saxBlobSuffixProperty: String = ".blob"
+
+  private val featureMap = mutable.Map[String, Boolean](saxNamespaceFeature -> false,
+    saxNamespacePrefixFeature -> false)
+
+  override def getFeature(name: String): Boolean = {
+    if (name == saxNamespaceFeature || name == saxNamespacePrefixFeature) {
+      featureMap(name)
+    } else {
+      throw new SAXNotRecognizedException("Feature unsupported: " + name + ".\n" +
+        "Supported features are: " + featureMap.keys.mkString(", "))
+      false
+    }
+  }
+
+  override def setFeature(name: String, value: Boolean): Unit = {
+    if (name == saxNamespaceFeature || name == saxNamespacePrefixFeature) {
+      featureMap(name) = value
+    } else {
+      throw new SAXNotRecognizedException("Feature unsupported: " + name + ".\n" +
+        "Supported features are: " + featureMap.keys.mkString(", "))
+    }
+  }
+
+  override def getProperty(name: String): AnyRef = {
+    val expectedUniqueURNRegex = s"(${XMLUtils.DAFFODIL_SAX_URN_ROOT}:)(\\w+)".r
+    val prop = name match {
+      case expectedUniqueURNRegex(_, pr) if pr == "ParseResult" => saxParseResultProperty
```

Comment:
I think it would be more clear to just do string matching. E.g.
```scala
name match {
 case XMLUtils.DAFFODIL_SAX_PROPERTY_PASE_RESULT => saxParseResultProperty
...
}
```
The regex just adds extra complexity. Same with setProperty.